src/components/structures/RoomDirectory.tsx

```diff
@@ -350,44 +351,23 @@ export default class RoomDirectory extends React.Component<IProps, IState> {
     };
 
     private onJoinFromSearchClick = (alias: string) => {
-        // If we don't have a particular instance id selected, just show that rooms alias
-        if (!this.state.instanceId || this.state.instanceId === ALL_ROOMS) {
-            // If the user specified an alias without a domain, add on whichever server is selected
-            // in the dropdown
-            if (alias.indexOf(':') == -1) {
-                alias = alias + ':' + this.state.roomServer;
-            }
-            this.showRoomAlias(alias, true);
-        } else {
-            // This is a 3rd party protocol. Let's see if we can join it
-            const protocolName = protocolNameForInstanceId(this.protocols, this.state.instanceId);
-            const instance = instanceForInstanceId(this.protocols, this.state.instanceId);
-            const fields = protocolName
-                ? this.getFieldsForThirdPartyLocation(alias, this.protocols[protocolName], instance)
-                : null;
-            if (!fields) {
-                const brand = SdkConfig.get().brand;
-                Modal.createTrackedDialog('Unable to join network', '', ErrorDialog, {
-                    title: _t('Unable to join network'),
-                    description: _t('%(brand)s does not know how to join a room on this network', { brand }),
+        const cli = MatrixClientPeg.get();
+        try {
+            joinRoomByAlias(cli, alias, {
+                instanceId: this.state.instanceId,
+                roomServer: this.state.roomServer,
+                protocols: this.protocols,
+                metricsTrigger: "RoomDirectory",
+            });
+        } catch (e) {
+            if (e instanceof GenericError) {
+                Modal.createTrackedDialog(e.message, '', ErrorDialog, {
+                    title: e.message,
+                    description: e.description,
                 });
```

Comment:
Maybe `utils/error.ts` should offer this as `showErrorModal()`? I assume this could become a common pattern.

Reply:
I'm not sure we can do this as the `.ts` files should not have anything to do with UI according to our new style guide.
So i guess we should probably have some UI helpers?